module/users/helper.js

```diff
@@ -300,25 +351,49 @@ module.exports = class UserHelper {
                             solutionData => solutionData.externalId === solution.externalId
                         );
 
-                        if( solutionIndex < 0 ) {
+                        if (solutionIndex < 0) {
 
-                            let solutionOrObservationInformation = 
-                            users[user].isObservation ? 
-                            _observationInformation(
-                                program,
-                                users[user],
-                                solution
-                            ) : _solutionInformation(program,solution); 
+                            let solutionOrObservationInformation;
                             
-                            solutionOrObservationInformation["entities"] = [];
+                            if (users[user].isObservation) {
+                                solutionOrObservationInformation = _observationInformation
+                                (
+                                    program,
+                                    users[user],
+                                    solution
+                                )
+                            }
+                            else if (users[user].isSurvey) {
+                                solutionOrObservationInformation = _surveyInformation
```

Comment:
@deepakr123 - Please change variable name to what is relevant

Reply:
resolved